sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/Counter.java

```diff
@@ -327,8 +327,16 @@ public AggregationKind getKind() {
 
   /**
    * Returns if the counter contains non-committed aggregate.
+   *
+   * <p>After all possible mutations have completed, the reader should check
```

Comment:
This looks great. Can you put the details of this counter lifecycle/contract in the Counter class javadoc?